src/ICSharpCode.SharpZipLib/ICSharpCode.SharpZipLib.csproj

```diff
@@ -33,8 +33,18 @@ Please see https://github.com/icsharpcode/SharpZipLib/wiki/Release-1.3.3 for mor
 	<ItemGroup>
 		<PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All" />
 	</ItemGroup>
-	 
-	 <ItemGroup>
+
+	<ItemGroup Condition=" '$(TargetFramework)' == 'net45' ">
+		<PackageReference Include="System.Memory" Version="4.5.4" />
+		<PackageReference Include="System.Threading.Tasks.Extensions" Version="4.5.2" />
+	</ItemGroup>
+
+	<ItemGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
+		<PackageReference Include="System.Memory" Version="4.5.4" />
+		<PackageReference Include="System.Threading.Tasks.Extensions" Version="4.5.2" />
+	</ItemGroup>
```

Comment:
There's been a core goal of keeping the library dependency free. But if it means keeping .NET 4.5 compatibility without sacrificing newer features and optimizations, maybe this ought to be reconsidered...

Reply:
https://devblogs.microsoft.com/dotnet/dotnet-framework-45-46-461-end-of-support/ maybe we should consider net462 minimum?

Reply:
We are netstandard2.0 on ICSharpCode.Decompiler - which means net472+ compat for the old full framework. In 2022, it should be ok to ditch anything older than that and use the polyfills.

Reply:
Is the conclusion to remove the `net45` target and keep the polyfill packages? Those packages provide `ValueTask` and ArrayPool / Span / Memory.  

I think the addition of `ValueTask` is important from a performance perspective as without it the new implementation would perform quite a bit worse than the existing one for sync calls, which is obviously not great.

It's probably possible to remove the dependency on `System.Memory` (without sacrificing too much) by going back to using plain old `byte[]` and pulling an implementation of `ArrayPool` into this code base (similar to the very basic `StringBuilderPool` I have already added). Though my preference would be to keep this package reference too if you are happy with that?

Reply:
Additionally, if we are dropping .NET 4.5 support, we can just stick to `netstandard2.0` as it's supported by 4.6.1-4.8:
https://docs.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-2-0

I will make a new issue for this.